### PR TITLE
Cleanup Screens\Menu and fix cancelation token

### PR DESCRIPTION
**Pull request by 73794449**
A bit cleanup and fix for bag(if very fast click on some button and click esc may crash game because cancelation token was != null).

### CODE REVIEW

File: osu.Game/Rulesets/UI/DrawableRuleset.cs [hunk 1]
```diff
@@ -173,7 +172,10 @@ private void load(OsuConfigManager config, CancellationToken? cancellationToken)
 
             applyRulesetMods(mods, config);
 
-            loadObjects(cancellationToken);
+            if (cancellationToken == null)
+            {
```
Comment (iiSaLMaN) [inline osu.Game/Rulesets/UI/DrawableRuleset.cs]: unnecessary bracket

File: osu.Game/Rulesets/UI/DrawableRuleset.cs [hunk 2]
```diff
@@ -101,7 +100,7 @@ internal bool FrameStablePlayback
         private OnScreenDisplay onScreenDisplay;
 
         /// <summary>
-        /// Creates a ruleset visualisation for the provided ruleset and beatmap.
+        /// Creates a ruleset visualization for the provided ruleset and beatmap.
```
Comment (iiSaLMaN) [inline osu.Game/Rulesets/UI/DrawableRuleset.cs]: why did you change that?

File: osu.Game/Screens/Menu/ButtonArea.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class ButtonArea : Container, IStateful<Visibility>
         private readonly ButtonAreaBackground buttonAreaBackground;
         private Visibility state;
 
-        public const float BUTTON_AREA_HEIGHT = 100;
+        public const int BUTTON_AREA_HEIGHT = 100;
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Menu/ButtonArea.cs]: Why? Height is of type `float`, not `int`

File: osu.Game/Screens/Menu/ButtonArea.cs [hunk 4]
```diff
@@ -72,6 +72,7 @@ public ButtonSystemState ButtonSystemState
         public Visibility State
         {
             get => state;
+
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Menu/ButtonArea.cs]: inconsistent with the code style, we have the setter in a new line after getter:
```csharp
get => ...
set => ...
```
(it's already obvious i don't know why you changed it)

File: osu.Game/Screens/Menu/MenuSideFlashes.cs [hunk 5]
```diff
@@ -102,7 +102,7 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
 
         private void flash(Drawable d, double beatLength, bool kiai, TrackAmplitudes amplitudes)
         {
-            d.FadeTo(Math.Max(0, ((d.Equals(leftBox) ? amplitudes.LeftChannel : amplitudes.RightChannel) - amplitude_dead_zone) / (kiai ? kiai_multiplier : alpha_multiplier)), box_fade_in_time)
+            d.FadeTo(System.Math.Max(0, ((d.Equals(leftBox) ? amplitudes.LeftChannel : amplitudes.RightChannel) - amplitude_dead_zone) / (kiai ? kiai_multiplier : alpha_multiplier)), box_fade_in_time)
```
Comment (EVAST9919) [inline osu.Game/Screens/Menu/MenuSideFlashes.cs]: Not sure about this

File: osu.Game/Screens/Menu/IntroScreen.cs [hunk 6]
```diff
@@ -109,7 +110,9 @@ protected override void LogoArriving(OsuLogo logo, bool resuming)
 
         protected void PrepareMenuLoad()
         {
-            LoadComponentAsync(mainMenu = new MainMenu());
```
Comment (EVAST9919) [inline osu.Game/Screens/Menu/IntroScreen.cs]: What's the difference?

File: osu.Game/Screens/Menu/ButtonSystem.cs [hunk 7]
```diff
@@ -240,6 +240,7 @@ private bool onOsuLogo()
         private ButtonSystemState state = ButtonSystemState.Initial;
 
         public override bool HandleNonPositionalInput => state != ButtonSystemState.Exit;
+
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Menu/ButtonSystem.cs]: why..., please take a look at the project's code style, we normally have properties/functions/etc.. together without a new line if they're a 1 line and related in some way (e.g `OnMouseDown`, `OnMouseUp`)

File: osu.Game/Screens/Menu/IntroCircles.cs [hunk 8]
```diff
@@ -61,8 +61,8 @@ private void load(OsuConfigManager config, BeatmapManager beatmaps, Framework.Ga
                 welcome = samples.Get(@"welcome");
         }
 
-        private const double delay_step_one = 2300;
-        private const double delay_step_two = 600;
+        private const int delay_step_one = 2300;
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Menu/IntroCircles.cs]: same here, why is it `int` when it's usage requires type `double` (e.g `IntroCircles:L89`)

File: osu.Game/Screens/Menu/MenuSideFlashes.cs (hunk 5 above)
Comment (T3RY4) [reply osu.Game/Screens/Menu/MenuSideFlashes.cs]: using System for one time

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/UI/DrawableRuleset.cs [hunk 9]
```diff
@@ -173,7 +172,10 @@ private void load(OsuConfigManager config, CancellationToken? cancellationToken)
 
             applyRulesetMods(mods, config);
 
-            loadObjects(cancellationToken);
+            if (cancellationToken == null)
```
Comment (EVAST9919) [inline osu.Game/Rulesets/UI/DrawableRuleset.cs]: Null check happens in line 192

**Comment by iiSaLMaN** [comment]
I highly suggest you take a look at how code is written in this project.

File: osu.Game/Screens/Menu/ButtonArea.cs (hunk 3 above)
Comment (EVAST9919) [reply osu.Game/Screens/Menu/ButtonArea.cs]: I'm using int for consts as well, so no judging here

File: osu.Game/Screens/Menu/MenuSideFlashes.cs (hunk 5 above)
Comment (EVAST9919) [reply osu.Game/Screens/Menu/MenuSideFlashes.cs]: We suggest move all the usings in using section, even they are used once

File: osu.Game/Screens/Menu/IntroScreen.cs (hunk 6 above)
Comment (T3RY4) [reply osu.Game/Screens/Menu/IntroScreen.cs]: Any big differences, but I think that it would be better for read.

File: osu.Game/Rulesets/UI/DrawableRuleset.cs (hunk 2 above)
Comment (T3RY4) [reply osu.Game/Rulesets/UI/DrawableRuleset.cs]: Its global version of English

File: osu.Game/Screens/Menu/MenuSideFlashes.cs (hunk 5 above)
Comment (T3RY4) [reply osu.Game/Screens/Menu/MenuSideFlashes.cs]: Ok

File: osu.Game/Screens/Menu/ButtonArea.cs (hunk 4 above)
Comment (T3RY4) [reply osu.Game/Screens/Menu/ButtonArea.cs]: Srry, bad settings of autocorrection. When I would at home, I change it

File: osu.Game/Rulesets/UI/DrawableRuleset.cs (hunk 2 above)
Comment (iiSaLMaN) [reply osu.Game/Rulesets/UI/DrawableRuleset.cs]: British English is mostly used here.

**Review by frenzibyte — CHANGES_REQUESTED** [review]
Did you test this locally? I'm pretty sure you didn't. What if the cancellation token is not null? the objects will never be loaded and that would simply result in blank gameplay.
